### ISSUE TITLE
Pipwire issue

### ISSUE THREAD

**Issue by s02sv650s** [awawa-dev/HyperHDR#343]

**Comment by s02sv650s** [comment]
I've been having some wicked problems getting this to work.  Worst part is I don't know what is preventing this from working?  I look over my work constantly and I can't see what I have wrong.  I had it working yesterday for about 2 hours, it worked flawlessly.  I left the house, and when I came back it wasn't working again.  I hope someone can shed some light.  Maybe the LEDs have some sort of a defect with them?  What is this pipewire error I keep getting?

**Comment by awawa-dev** [comment]
The pipewire error is irrelevant: your system doesn't have installed pipewire 0.3 so no pipewire grabber. But USB grabber works (and preview confirms it) and is sending frames. One of the reason why driving ws2801x directly from Rpi is not supported by HyperHDR is almost complete lack of diagnostic protocol when something goes wrong, the second reason is missing 3.3V to 5V voltage level shifter... maybe a cable is disconnected or Rpi GPIO is fried. Anyway your ws2801x library doesnt report any error.

**Comment by s02sv650s** [comment]
Is there a way I can test the GPIO to see if there's anything wrong with them?  Some sort of a command for the pi to run a test on them, or do I have to breakout the volt meter, haha.

**Comment by awawa-dev** [comment]
Without an oscilloscope is impossible to diagnose the GPIO, logic state analyzer would show something but without telling the voltage level but for SPI not for emulating ws2801x protocol which is a mess, the volt meter is not helpful. There are some script like https://github.com/rm-hull/spidev-test but again, there are made to test SPI interfaces and I have never tested them.

**Comment by s02sv650s** [comment]
Is it possible to change the GPIO value in HyperHDR.  Like to go from GPIO 18 to GPIO 19, since 19 still does PWM.

**Comment by awawa-dev** [comment]
Don't know and haven't tested it but you can try it, I removed that module from the official support as it's stated in the FAQ and I recommend to use external ESP. This approach is roulette that sometimes works and sometimes doesn't. And it certainly goes against the technical requirements of the LED strip.

**Comment by s02sv650s** [comment]
> Don't know and haven't tested it but you can try it, I removed that module from the official support as it's stated in the FAQ and I recommend to use external ESP. This approach is roulette that sometimes works and sometimes doesn't. And it certainly goes against the technical requirements of the LED strip.

So, you recommend going from the raspberry pi to an ESP, instead of plugging the Raspberry Pi directly into the LED strips?
Sorry for my ignorance, I'm just trying to learn and understand this.

**Comment by awawa-dev** [comment]
Yes, optimal setup is using ESP and the level shifter. For cable solution for Esp you have HyperSPI, HyperSerialEsp8266, HyperSerialEsp32 and for wireless WLED. For a level shifter use SN74AHCT125N https://github.com/awawa-dev/HyperHDR/wiki/Level-Shifter . But before changing the setup make sure it's not a cable issue or broken LED strip power supply: measure the +5V voltage on the LED strip. Do you hame a commong ground between Rpi and the LED strip?

**Comment by s02sv650s** [comment]
Yes, there is a common ground.  The RPI and the Strips are hooked up to the same ground.
I watched a guide on youtube for this, and that person didn't use an ESP.  I didn't even hear about an ESP until I started poking around here.

**Comment by awawa-dev** [comment]
> I watched a guide on youtube for this, and that person didn't use an ESP.

https://github.com/awawa-dev/HyperHDR/discussions/320 😉

**Comment by awawa-dev** [comment]
Thread moved to the discussion.
